### DISCUSSION TITLE
How to setup a SOCKS5 Tunnel for private browsing with SSH?

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1532] [Q&A] [answered]
Does Blink support SOCKS5 over ssh?

**Comment by carloscabanero** [comment, marked as answer]
SOCKS proxy can be used as a simple VPN to browse the internet or connect with back-end services behind a Firewall.

You can configure iOS to route all traffic through Blink Shell Proxy. Let's open: **Settings** > **Wi-Fi**, select your network, and click on **Connection Information**.

![IMG_2393.jpeg](https://res.craft.do/user/full/0ef94637-30d5-80f2-9fa1-0089b763dcf6/45052B07-8AE8-49F1-817F-5D966D7393D0_2/YCSZhiFqgyARB0t20nkVD7XtIK8hNOKCYBqNfJ7nWOcz/IMG_2393.jpeg)

Scroll down to the **HTTP Proxy** panel, and click **Configure Proxy**.

![IMG_2395.jpeg](https://res.craft.do/user/full/0ef94637-30d5-80f2-9fa1-0089b763dcf6/768A7CD5-522C-4D3A-B8B7-5E1F8FF17F08_2/Sr2JpVjMRKTLHvXdqHkpZ9nOzh8yrDlccNPxOzM2Q1Yz/IMG_2395.jpeg)

Select **Automatic**, and paste the [https://blink.sh/proxy.pac](https://blink.sh/proxy.pac) link.

![IMG_2399.jpeg](https://res.craft.do/user/full/0ef94637-30d5-80f2-9fa1-0089b763dcf6/36241588-39E6-4CEA-8062-F24331D2DBEF_2/H47ZvoMA9IwWh7kwsN1r2o8ByAJbt7KIhUhieAEYc6sz/IMG_2399.jpeg)

To start dynamic port forwarding inside Blink Shell, just type the usual `ssh -D 10800`. As this is an SSH tunnel, we recommend using `geo` track in Blink Shell to ensure that the connection is locked while you are browsing.

The usual SOCKS5 port is **1080**, but it is taken by iOS. Our own [blink.sh/proxy.pac](http://blink.sh/proxy.pac) file configures the listener on port **10800**. If you like to use a different port, you need to use your own proxy.pac file, here's an example.

```javascript
function FindProxyForURL(url, host)
{ 
 return "SOCKS localhost:10800";
}
```

We suggest creating a Gist with that configuration and linking to it instead.

### More information

- SOCKS Proxy Primer: [https://securityintelligence.com/posts/socks-proxy-primer-what-is-socks5-and-why-should-you-use-it/](https://securityintelligence.com/posts/socks-proxy-primer-what-is-socks5-and-why-should-you-use-it/)
- MDN article about Proxy Auto-Configuration (PAC) file: [https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_PAC_file](https://developer.mozilla.org/en-US/docs/Web/HTTP/Proxy_servers_and_tunneling/Proxy_Auto-Configuration_PAC_file)

**Reply by martinpaljak** [reply]
Some solid instructions on how to set up socks proxy with mobile data connection (not wifi) would be appreciated.
